lib/js/atoms/SurveyToggle/SurveyToggle.vue

```diff
@@ -0,0 +1,296 @@
+		}
+	}
+
+	&.-defaultPrimary &__ring {
+		border-color: $color-primary;
+	}
+
+	&.-selectedPrimary &__ring {
+		border-color: $color-total-white;
+	}
+
+	&.-selectedNeutral &__ring {
+		border-color: $color-total-white;
+	}
+
+	&.-defaultPrimary.-disabled &__ring {
+		border-color: $color-primary-disabled;
+	}
+
+	// content
+	&__content {
+		@include headlineXS();
+		@include textBold();
+
+		color: $color-minor-supporting;
+	}
+
+	&.-defaultPrimary &__content {
+		color: $color-primary;
+
+		.-hovered & {
+			$border-color: $color-primary-hovered;
+		}
+	}
+
+	&.-defaultPrimary.-disabled &__content {
+		color: $color-primary-disabled;
+	}
+
+	// icon
+	&__icon {
+		color: $color-total-white;
+		height: $icon-xs;
+		width: $icon-xs;
+	}
+
+	// label
+	&__label {
+		@include textInfoS();
+		@include textBold();
+
+		color: $color-minor;
+		text-align: center;
+		margin-top: $space-xxs;
+		min-height: 2em;
+		max-width: 100%;
+
+		@media #{breakpoint-s()} {
+			@include textInfoM();
+
+			font-weight: normal;
+		}
+	}
+}
+</style>
+
+<script lang="ts">
+import {
+	SURVEY_TOGGLE_MEANINGS,
+	SURVEY_TOGGLE_STATES,
+	SURVEY_TOGGLE_STATUSES,
+} from './SurveyToggle.consts';
+import DsIcon, { ICON_SIZES, ICONS } from '../Icon';
+import Ripple from 'vue-ripple-directive';
+
+export default {
+	name: 'SurveyToggle',
+	components: {
+		DsIcon,
+	},
+	directives: {
+		ripple: Ripple,
+	},
+	props: {
+		label: {
+			type: String,
+			default: null,
+		},
+		contentText: {
+			type: String,
+			default: null,
+		},
+		meaning: {
+			type: String,
+			default: SURVEY_TOGGLE_MEANINGS.PRIMARY,
+			validate(meaning) {
+				return Object.values(SURVEY_TOGGLE_MEANINGS).includes(meaning);
+			},
+		},
+		status: {
+			type: String,
+			default: SURVEY_TOGGLE_STATUSES.DEFAULT,
+			validate(status) {
+				return Object.values(SURVEY_TOGGLE_STATUSES).includes(status);
+			},
+		},
+		state: {
+			type: String,
+			default: SURVEY_TOGGLE_STATES.DEFAULT,
+			validate(state) {
+				return Object.values(SURVEY_TOGGLE_STATES).includes(state);
+			},
+		},
+		contentIcon: {
+			type: Object,
+			default() {
+				return ICONS.FA_CHECK_SOLID;
+			},
+			validate(contentIcon) {
+				return Object.values(ICONS).includes(contentIcon);
+			},
+		},
+	},
+	data() {
+		return {
+			hovered: false,
+		};
+	},
+	computed: {
+		isSelectedPrimary() {
+			return (
+				this.meaning === SURVEY_TOGGLE_MEANINGS.PRIMARY &&
+				this.status === SURVEY_TOGGLE_STATUSES.SELECTED
+			);
+		},
+		isDefaultPrimary() {
+			return (
+				this.meaning === SURVEY_TOGGLE_MEANINGS.PRIMARY &&
+				this.status === SURVEY_TOGGLE_STATUSES.DEFAULT
+			);
+		},
+		isSelectedNeutral() {
+			return (
+				this.meaning === SURVEY_TOGGLE_MEANINGS.NEUTRAL &&
+				this.status === SURVEY_TOGGLE_STATUSES.SELECTED
+			);
+		},
+		isHoveredState() {
+			if (this.state === SURVEY_TOGGLE_STATES.DISABLED) {
+				return false;
+			}
+
+			if (this.state === SURVEY_TOGGLE_STATES.HOVERED) {
+				return true;
+			}
```

Comment:
This is only for Storybook, right? I don't like this approach, it bloats the API unnecessarily.

Reply:
We have a separate discussion with UX about it: https://www.figma.com/file/izQdYyiBR1GQgFkaOIfIJI?node-id=1055:23487#119256600. I already have a PoC here https://github.com/bethinkpl/design-system/pull/73, but there're some problems: https://github.com/chromaui/storybook-addon-pseudo-states/issues/6

Can't handle all DS problems while developing new features, so I suggest leaving it for now